File: solutions/java/src/restaurantmanagementsystem/payment/Payment.java
package restaurantmanagementsystem.payment;

public class Payment {
    private final int id;
    private final double amount;
    private final PaymentMethod method;
    private final PaymentStatus status;

    public Payment(int id, double amount, PaymentMethod method, PaymentStatus status) {
        this.id = id;
        this.amount = amount;
        this.method = method;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public double getAmount() {
        return amount;
    }

    public PaymentMethod getMethod() {
        return method;
    }

    public PaymentStatus getStatus() {
        return status;
    }
}

File: solutions/csharp/movieticketbookingsystem/Models/Payment.cs
class Payment
{
    private readonly string id;
    private readonly double amount;
    private readonly PaymentStatus status;
    private readonly string transactionId;

    public Payment(double amount, PaymentStatus status, string transactionId)
    {
        this.id = Guid.NewGuid().ToString();
        this.amount = amount;
        this.status = status;
        this.transactionId = transactionId;
    }

    public PaymentStatus GetStatus() { return status; }
}